File: Golang/Relative Ranks.go
//1
package main

import (
	"sort"
	"strconv"
)

type MyInt struct {
	Val int
	Index int
}

func findRelativeRanks(nums []int) []string {
	athletes := make([]MyInt, len(nums))
	for i, num := range nums {
		athletes[i].Index=i
		athletes[i].Val=num
	}
	sort.Slice(athletes, func(i, j int) bool {
		return athletes[i].Val>athletes[j].Val
	})
	rank := make([]string, len(nums))
	for i, athlete := range athletes {
		switch i {
		case 0:
			rank[athlete.Index]="Gold Medal"
		case 1:
			rank[athlete.Index]="Silver Medal"
		case 2:
			rank[athlete.Index]="Bronze Medal"
		default:
			rank[athlete.Index] = strconv.Itoa(i+1)
		}
	}
	return rank

}

//2
func findRelativeRanks(score []int) []string {
    var temp []int
    mp := map[int]int{}
    var ans []string

    temp = append(temp, score...)

    sort.Sort(sort.Reverse(sort.IntSlice(temp)))
    for i := range temp{
        mp[temp[i]] = i+1
    }
    
    for i := range score{
        if mp[score[i]] == 1{
            ans = append(ans, "Gold Medal")
        }else if mp[score[i]] == 2 {
            ans = append(ans, "Silver Medal")
        }else if mp[score[i]] == 3{
            ans = append(ans,  "Bronze Medal")
        }else{
            num := strconv.Itoa(mp[score[i]])
            ans = append(ans, num)
        }
    }

    return ans
}

File: Python3/Relative Ranks.py
#1
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
         # Prepare a medal list, an empty result list and an empty dictionary
        Medal = ["Gold Medal", "Silver Medal", "Bronze Medal"]
        size = len(score)
        res = [""]*size
        my_map = {}
        
        # Set the dictionary
        # Key is the number, value is the index of the number in the orginal input list
        for i in range(size):
            my_map[score[i]] = i
            
        # Sort the input list in the descending order
        score.sort(reverse=True)
        
        # Set the result list using the sorted input list and the dictionary
        for i in range(size):
            if i < 3:
                res[my_map[score[i]]] = Medal[i]            
            else:
                res[my_map[score[i]]] = str(i+1)
                
        return res

#2
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        rankings = []
        for i, val in enumerate(score):
            heappush(rankings, (-val, i))
        ans = [''] * len(score)
        r = 1
        rank = ["Gold Medal", "Silver Medal", "Bronze Medal"]
        while len(rankings) != 0:
            _, i = heappop(rankings)
            if r <= 3:
                ans[i] = rank[r-1]
            else:
                ans[i] = f'{r}'
            r += 1
        return ans